neurox/interpretation/linear_probe.py

```diff
@@ -381,6 +392,8 @@ class and their associated scores are also part of the dictionary.
 
     if use_gpu:
         probe = probe.cuda()
+        if X.dtype == 'float16': # cuda, fp16 in X: perform eval in half precision
```

Comment:
So it seems like we can do evaluation in half precision but not training; If you think this gives us a reasonable speed up, we can leave this in; If not, we can just do full precision eval just like in train; I'll defer to your expertise here

Reply:
The problem is in the current version of torch: On CPU, you should not use float16 for various reasons (see e.g. [this thread](https://github.com/pytorch/pytorch/issues/48245)). This is why CPU is treated differently than GPU here. I clarified the method comment.

Reply:
actually my question was at a higher level; given that we already do full precision training, is the benefit we get from "half precision evaluation on a GPU only" worth the added code complexity?

If you think that it is worth it, we can leave it in!